jni/src/common/Foreign/JNI/String.hs

```diff
@@ -66,6 +66,7 @@ toByteString (String bs) = BS.init bs
 -- a null-terminated buffer first.
 fromByteString :: ByteString -> String
 fromByteString bs
+  | bs == BS.empty = String $ BS.singleton 0
```

Comment:
Isn't this equivalent to the otherwise case?

`BS.singleton 0` looks equivalent to `BS.snoc BS.empty 0`

Reply:
The right-hand side is indeed equivalent, but the `otherwise` case is never reached if bs is empty:
`BS.last BS.empty` raises a runtime error

Reply:
Though I suppose it could also be fixed by replacing the first guard with `BS.length bs > 0 && BS.last bs == 0`